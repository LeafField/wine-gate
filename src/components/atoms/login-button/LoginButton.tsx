import React from "react";
import Link from "next/link";

const LoginButton = () => {
  return (
    <Link
      className="w-full rounded-xl border border-gray py-2 text-center transition-colors duration-300 hover:border-blue focus:outline-blue"
      href={"/login"}
    >
      ログイン/会員登録
    </Link>
  );
};

export default LoginButton;
