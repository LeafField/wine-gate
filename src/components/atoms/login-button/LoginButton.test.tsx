import { screen, render } from "@testing-library/react";
import LoginButton from "./LoginButton";

describe("LoginButtonの単体テスト", () => {
  test("LoginButtonが表示されているか", () => {
    render(<LoginButton />);
    expect(screen.getByText("ログイン/会員登録")).toBeInTheDocument();
  });

  test("スナップショットテスト", () => {
    const { container } = render(<LoginButton />);
    expect(container).toMatchSnapshot();
  });
});
