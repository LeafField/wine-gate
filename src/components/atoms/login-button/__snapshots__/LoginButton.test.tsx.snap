// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButtonの単体テスト スナップショットテスト 1`] = `
<div>
  <a
    class="w-full rounded-xl border border-gray py-2 text-center transition-[border-color] duration-300 hover:border-blue focus:outline-blue"
    href="/login"
  >
    ログイン/会員登録
  </a>
</div>
`;
